browser-test/src/civiform_admin_question_list.test.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  AdminPrograms,
+  AdminQuestions,
+  createTestContext,
+  loginAsAdmin,
+} from './support'
+
+describe('Most recently updated question is at top of list.', () => {
+  const ctx = createTestContext()
+  it('sorts by last updated, preferring draft over active', async () => {
+    const {page, adminPrograms, adminQuestions} = ctx
+
+    await loginAsAdmin(page)
+
+    const questionOnePublishedText = 'question list test question one'
+    const questionTwoPublishedText = 'question list test question two'
+    await adminQuestions.addNameQuestion({
+      questionName: questionOnePublishedText,
+      questionText: questionOnePublishedText,
+    })
+    await adminQuestions.addNameQuestion({
+      questionName: questionTwoPublishedText,
+      questionText: questionTwoPublishedText,
+    })
+
+    // Note: CI tests already have test questions
+    // available. As such, we only assert the order
+    // of the questions added in this test.
+    // TODO(#3029): Consider asserting on the whole list if it can be considered stable even in
+    // prober environments and with canonical question creation.
+
+    // A question cannot be published in isolation. In order to allow making created questions
+    // active, create a fake program.
+    const programName = 'question list test program'
+    await adminPrograms.addProgram(programName)
+
+    // Most recently added question is on top.
+    await expectQuestionListTopElements(adminQuestions, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+    await expectQuestionBankTopElements(programName, adminPrograms, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+
+    // Publish all programs and questions, order should be maintained.
+    await adminPrograms.publishProgram(programName)
+    // Create a draft version of the program so that the question bank can be accessed.
+    await adminPrograms.createNewVersion(programName)
+    await expectQuestionListTopElements(adminQuestions, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+    await expectQuestionBankTopElements(programName, adminPrograms, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+
+    // Now create a draft version of the previously last question. After,
+    // it should be on top.
+    await adminQuestions.createNewVersion(questionOnePublishedText)
+    const questionOneDraftText = `${questionOnePublishedText} new version`
```

Comment:
Here "new version" implicitly comes from the `createNewVersion` function?

Reply:
That's correct. This was some surprising action-at-a-distance. I've now added an explicit comment around that.